test/extended/images/helper.go

```diff
@@ -223,40 +227,122 @@ func BuildAndPushImageOfSizeWithDocker(
 
 	if shouldSucceed {
 		if err != nil {
-			return "", fmt.Errorf("Got unexpected push error: %v", err)
+			return "", "", fmt.Errorf("Got unexpected push error: %v", err)
 		}
 		if len(imageDigest) == 0 {
 			match := rePushedImageDigest.FindStringSubmatch(out)
 			if len(match) < 2 {
-				return imageDigest, fmt.Errorf("Failed to parse digest")
+				return imageDigest, "", fmt.Errorf("Failed to parse digest")
 			}
 			imageDigest = match[1]
 		}
 		return
 	}
 
 	if err == nil {
-		return "", fmt.Errorf("Push unexpectedly succeeded")
+		return "", "", fmt.Errorf("Push unexpectedly succeeded")
 	}
 	if !reExpectedDeniedError.MatchString(err.Error()) {
-		return "", fmt.Errorf("Failed to match expected %q in: %q", reExpectedDeniedError.String(), err.Error())
+		return "", "", fmt.Errorf("Failed to match expected %q in: %q", reExpectedDeniedError.String(), err.Error())
 	}
 
-	return "", nil
+	return "", "", nil
```

Comment:
Maybe it could be better structured. Nevertheless, the returned values are not wrong. This line is reached when a push failure is expected and the push really failed - so there's no usable output for the caller. Well, maybe `imageID` is of some use unless `removeAfterPush` is `true`.